Dashboard/app/js/lib/views/maps/map-views-common.js

```diff
@@ -180,307 +164,132 @@ FLOW.NavMapsView = FLOW.View.extend({
+            delete self.mediaMarkers[filename];
+            delete self.mediaMarkerSelected[filename];
           }
         }
+      });
+  },
 
-        if (mapExists) {
-          //overlay named map
-          self.createLayer(namedMapObject.mapObject, namedMapObject.mapName, "");
-        }else{
-          //create new named map
-          self.namedMaps(
-            namedMapObject.mapObject,
-            namedMapObject.mapName,
-            namedMapObject.tableName,
-            namedMapObject.query,
-            namedMapObject.interactivity);
-        }
+  surveySelection: function () {
+      this.clearMap();
+      if (!Ember.none(this.get('selectedSurvey'))) {
+          FLOW.mapsController.loadNamedMap(this.selectedSurvey.get('keyId'));
       }
-    });
-  },
+  }.observes('this.selectedSurvey'),
+
+  surveyGroupSelection: function () {
+      this.clearMap();
+      FLOW.mapsController.clearCartodbLayer();
+  }.observes('FLOW.selectedControl.selectedSurveyGroup'),
 
   /**
     Helper function to dispatch to either hide or show details pane
   */
   handleShowHideDetails: function () {
     if (this.detailsPaneVisible) {
-      this.hideDetailsPane();
+      this.set('detailsPaneVisible', false);
     } else {
-      this.showDetailsPane();
+      this.set('detailsPaneVisible', true);
     }
   },
 
   /**
-    Slide in the details pane
-  */
-  showDetailsPane: function () {
-    var button;
-
-    button = this.$('#mapDetailsHideShow');
-    button.html(Ember.String.loc('_hide') + ' &rsaquo;');
-    this.set('detailsPaneVisible', true);
-
-    this.$('#flowMap').animate({
-      width: '75%'
-    }, 200);
-    this.$('#pointDetails').animate({
-      width: '24.5%'
-    }, 200).css({
-      overflow: 'auto',
-      marginLeft: '-2px'
-    });
-    this.$(this.detailsPaneElements, '#pointDetails').animate({
-      opacity: '1'
-    }, 200).css({
-      display: 'inherit'
-    });
-  },
-
-
-  /**
-    Slide out details pane
-  */
-  hideDetailsPane: function (delay) {
-    var button;
-
-    delay = typeof delay !== 'undefined' ? delay : 0;
-    button = this.$('#mapDetailsHideShow');
-
-    this.set('detailsPaneVisible', false);
-    button.html('&lsaquo; ' + Ember.String.loc('_show') );
-
-    this.$('#flowMap').delay(delay).animate({
-      width: '99.25%'
-    }, 200);
-    this.$('#pointDetails').delay(delay).animate({
-      width: '0.25%'
-    }, 200).css({
-      overflow: 'scroll-y',
-      marginLeft: '-2px'
-    });
-    this.$(this.detailsPaneElements, '#pointDetails').delay(delay).animate({
-      opacity: '0',
-      display: 'none'
-    });
-  },
-
-  /**
     If a placemark is selected and the details pane is hidden make sure to
     slide out
   */
   handlePlacemarkDetails: function () {
-    var details;
-
-    details = FLOW.placemarkDetailController.get('content');
+    var details = FLOW.placemarkDetailController.get('content');
 
     if (!this.detailsPaneVisible) {
```

Comment:
you have this block `if (!this.detailsPane) {.... set to true}` which is repeated in a few places. I suggest to collapse them all under a single function `showDetailsPane()` attached to this view. That can make it more explicit what you are attempting to do with that block.

But It will be cleaner this time around because you are only using properties on the view rather than all the jQuery and HTML that was in the previous version.